core/base-service/got.js

```diff
@@ -0,0 +1,96 @@
+'use strict'
+
+const got = require('got')
+const { Inaccessible, InvalidResponse } = require('./errors')
+
+function requestOptions2GotOptions(options) {
```

Comment:
This allows us to pass a request-like options object to got without modifying service code (mostly).
I reckon once we've completely removed all our usage of request, we remove this translation layer, update the service code and update https://github.com/badges/shields/blob/master/doc/TUTORIAL.md to point to got's docs.